app/server/lib/AttachmentFileManager.ts

```diff
@@ -135,10 +142,54 @@ export class AttachmentFileManager extends EventEmitter {
     fileExtension: string,
     fileData: Buffer
   ): Promise<AddFileResult> {
-    const fileIdent = await this._getFileIdentifier(fileExtension, Readable.from(fileData));
+    const fileIdent = await this._getFileIdentifier(fileExtension, stream.Readable.from(fileData));
     return this._addFile(storeId, fileIdent, fileData);
   }
 
+  public async addMissingFileData(
```

Comment:
There are enough nuances in this method that it could do with a solid docstring.

Reply:
Added one, let me know if that conveys enough info.